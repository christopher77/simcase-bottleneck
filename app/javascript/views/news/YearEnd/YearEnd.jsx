import React from "react";
import yearEnd from "../../../assets/News/yearEnd.svg";
import styles from "./YearEnd.module.css";
import { navigate } from "@reach/router";
import ImageSvg from "../../../components/ui/ImageSvg";

function YearEnd(props) {
  const { success } = props;
  const getTotalScore = totalScore => props.setTotalScore(totalScore);
  const getHitsRate = () =>
    props.getSuccessRate(((success / 20) * 100).toFixed(2));

  function navigateToFinalScore() {
    getTotalScore();
    getHitsRate();
    navigate("/finalscore");
  }

  return (
    <div className={styles.screenContainer}>
      <ImageSvg src={yearEnd} alt="year-end-image" />
      <div className={styles.contentContainer}>
        <h4 className={styles.congrats}>congratulations</h4>
        <h4>The calendar year is at a close.</h4>
        <h4>
          See how your plant performed relative to your peers and decide if you
          want to try again
        </h4>
      </div>
      <button
        className={styles.resultsButton}
        onClick={navigateToFinalScore}
        autoFocus={true}
      >
        results
      </button>
    </div>
  );
}

export default YearEnd;
