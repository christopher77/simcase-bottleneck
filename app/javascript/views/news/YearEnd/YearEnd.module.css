h4 {
  font-weight: 500;
}

.screenContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: calc(100% - 40px);
  height: 80%;
  padding: 20px;
}

.congrats {
  text-transform: uppercase;
}

.resultsButton {
  background: #73c056;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 47px;
  color: black;
  width: 10em;
  height: 42px;
  padding: 8px 12px;
  text-transform: uppercase;
  font-size: 17px;
  font-weight: bold;
  border-color: transparent;
}

.resultsButton:active {
  transform: scale(1.1);
  box-shadow: 0px 8px 4px rgba(0, 0, 0, 0.25);
}

.contentContainer {
  height: 45%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: center;
  margin: 15px 0px;
}
