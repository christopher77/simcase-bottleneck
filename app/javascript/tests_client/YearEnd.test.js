import React from "react";
import { render } from "@testing-library/react";
import YearEnd from "../views/news/YearEnd/YearEnd";

test("YearEnd view", () => {
  const { asFragment } = render(<YearEnd />);
  expect(asFragment).toMatchSnapshot();
});
