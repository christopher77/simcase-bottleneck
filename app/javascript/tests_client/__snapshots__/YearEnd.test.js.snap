// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearEnd view 1`] = `[Function]`;
